### PR TITLE
Futurize Arkane to Py3

### PR DESCRIPTION
**Pull request by alongd**
Arkane Py3 changes, relocated from #1691

### CODE REVIEW

**Comment by amarkpayne** [comment]
@alongd I see that you have made changes to some of the files that I changed in my PR. Will this create conflicts? If so we should figure out how to incorporate all of the needed changes into 1 PR and drop the changes from the other (I am happy to get rid of the Arkane changes in my PR if need be).

**Comment by alongd** [comment]
@amarkpayne, I think this is finally ready for review

**Comment by alongd** [comment]
Thank you @oscarwumit and @amarkpayne for all the help!

File: arkane/molpro.py [hunk 1]
```diff
@@ -33,16 +28,23 @@
...
 ###############################################################################
 
+"""
+Arkane Molpro module
+Used to parse Molpro output files
+"""
+
 import math
-import numpy
 import logging
```
Comment (amarkpayne) [inline arkane/molpro.py]: Switch `logging` and `math`

File: arkane/pdep.py [hunk 2]
```diff
@@ -34,22 +34,22 @@
...
+
+from rmgpy.chemkin import writeKineticsEntry
+from rmgpy.exceptions import InvalidMicrocanonicalRateError, ModifiedStrongCollisionError, PressureDependenceError
 from rmgpy.kinetics import Chebyshev, PDepArrhenius
 from rmgpy.reaction import Reaction
 from rmgpy.kinetics.tunneling import Wigner, Eckart
 from rmgpy.data.kinetics.library import LibraryReaction
-from rmgpy.chemkin import writeKineticsEntry
-from rmgpy.exceptions import InvalidMicrocanonicalRateError, ModifiedStrongCollisionError
+import rmgpy.quantity as quantity
```
Comment (amarkpayne) [inline arkane/pdep.py]: Here too

File: arkane/qchem.py [hunk 3]
```diff
@@ -33,17 +28,24 @@
...
-import numpy
+
+import numpy as np
 
 import rmgpy.constants as constants
-from rmgpy.exceptions import InputError
 from rmgpy.statmech import IdealGasTranslation, NonlinearRotor, LinearRotor, HarmonicOscillator, Conformer
 
 from arkane.common import check_conformer_energy, get_element_mass
+from arkane.exceptions import LogError
```
Comment (amarkpayne) [inline arkane/qchem.py]: This is here twice

File: arkane/qchemTest.py [hunk 4]
```diff
@@ -28,15 +28,17 @@
 #                                                                             #
 ###############################################################################
 
+"""
+This module contains unit tests of the :mod:`arkane.qchem` module.
+"""
+
 import unittest
 import os
```
Comment (amarkpayne) [inline arkane/qchemTest.py]: Swap these

File: arkane/statmech.py [hunk 5]
```diff
@@ -34,36 +34,34 @@
...
 
 import rmgpy.constants as constants
+from rmgpy.exceptions import InputError, ElementError
+from rmgpy.molecule.molecule import Molecule
 from rmgpy.species import TransitionState, Species
-from rmgpy.statmech.translation import Translation, IdealGasTranslation
-from rmgpy.statmech.rotation import Rotation, LinearRotor, NonlinearRotor, KRotor, SphericalTopRotor
-from rmgpy.statmech.vibration import Vibration, HarmonicOscillator
-from rmgpy.statmech.torsion import Torsion, HinderedRotor, FreeRotor
 from rmgpy.statmech.conformer import Conformer
```
Comment (amarkpayne) [inline arkane/statmech.py]: Unused import. I think there are a couple below here as well

File: arkane/statmech.py [hunk 6]
```diff
@@ -688,7 +696,7 @@ def create_hindered_rotor_figure(self, angle, v_list, cosineRotor, fourierRotor,
         """
         try:
             import pylab
```
Comment (amarkpayne) [inline arkane/statmech.py]: Should we add `pylab` to our list of dependencies and conda environment files? If so we could move this to the general imports section and get rid of this code.

File: arkane/thermo.py [hunk 7]
```diff
@@ -33,28 +33,24 @@
...
-from rmgpy.thermo.wilhoit import Wilhoit
 from rmgpy.chemkin import writeThermoEntry
-from rmgpy.species import Species
+from rmgpy.exceptions import InputError
 from rmgpy.molecule import Molecule
 from rmgpy.molecule.util import retrieveElementCount
+from rmgpy.species import Species
+from rmgpy.statmech.rotation import LinearRotor, NonlinearRotor
+from rmgpy.thermo.wilhoit import Wilhoit
+from rmgpy.exceptions import InputError
```
Comment (amarkpayne) [inline arkane/thermo.py]: This is a duplicate import statement

File: arkane/common.py [hunk 8]
```diff
@@ -32,33 +28,36 @@
...
 from rmgpy.statmech.translation import IdealGasTranslation
 from rmgpy.statmech.vibration import HarmonicOscillator
-from rmgpy.pdep.collision import SingleExponentialDown
-from rmgpy.transport import TransportData
 from rmgpy.thermo import NASA, Wilhoit, ThermoData, NASAPolynomial
-from rmgpy.species import Species, TransitionState
-import rmgpy.constants as constants
+from rmgpy.transport import TransportData
 
 from arkane.pdep import PressureDependenceJob
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/common.py]: I think we should separate them just because we'd like to think of Arkane and RMG as different software, although I agree with PyCharm that technically it's the same. I consciously chose to separate them, but I'm open to different opinions

File: arkane/common.py [hunk 9]
```diff
@@ -317,31 +316,31 @@ def load_yaml(self, path, label=None, pdep=False):
 ################################################################################
 
 
-def is_pdep(jobList):
+def is_pdep(job_list):
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/common.py]: Thanks for double checking. I made sure that everywhere this function is used got updated. Let me know whether you'd like me to wait with that change.

File: arkane/exceptions.py [hunk 10]
```diff
@@ -46,3 +47,11 @@ class BondAdditivityCorrectionError(Exception):
     additivity corrections.
     """
     pass
+
+
+class LogError(Exception):
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/exceptions.py]: I prefer to keep LogError since it's related to the Log class

File: arkane/explorerTest.py [hunk 11]
```diff
@@ -28,19 +28,23 @@
...
 from arkane import Arkane
 from arkane.explorer import ExplorerJob
 
-
 ################################################################################
 
 
 @attr('functional')
-class testExplorerJob(unittest.TestCase):
+class TestExplorerJob(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/explorerTest.py]: OK. I think we should be safe with these kind of changes for Tests, but we could also hold off with that. Let me know what you think

File: arkane/gaussian.py [hunk 12]
```diff
@@ -33,17 +28,24 @@
...
 
-import rmgpy.constants as constants
+import numpy as np
+
 from rmgpy.statmech import IdealGasTranslation, NonlinearRotor, LinearRotor, HarmonicOscillator, Conformer
-from rmgpy.exceptions import InputError
+import rmgpy.constants as constants
 
 from arkane.common import check_conformer_energy, get_element_mass
+from arkane.exceptions import LogError
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/gaussian.py]: OK

File: arkane/gaussian.py [hunk 13]
```diff
@@ -465,6 +468,14 @@ def loadNegativeFrequency(self):
         frequencies.sort()
         frequency = [freq for freq in frequencies if freq < 0][0]
         if frequency is None:
-            raise Exception('Unable to find imaginary frequency of {1} '
-                            'in Gaussian output file {0}'.format(self.path, self.species.label))
+            raise LogError('Unable to find imaginary frequency of {1} '
+                           'in Gaussian output file {0}'.format(self.path, self.species.label))
         return frequency
+
+    def get_d1_diagnostic(self):
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/gaussian.py]: OK

File: arkane/kinetics.py [hunk 14]
```diff
@@ -392,7 +382,7 @@ def draw(self, output_directory, format='pdf'):
         KineticsDrawer().draw(self.reaction, format=format, path=path)
 
 
-class KineticsDrawer:
+class KineticsDrawer(object):
```
[the remark this answers is not in the document]
Comment (alongd) [reply arkane/kinetics.py]: Sorry, are you referring to the `(object)` addition? If its implicit in Py3, then I'm OK with removing it everywhere (we don't need it for the Py2 tests, it already passed them without it)

File: arkane/statmech.py (hunk 6 above)
Comment (alongd) [reply arkane/statmech.py]: Good point. Why are we using pyab and not matplotlib, anyway?

File: arkane/common.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (amarkpayne) [reply arkane/common.py]: I am fine with them being a separate block then

File: arkane/pdep.py [hunk 15]
```diff
@@ -40,13 +40,13 @@
 
 import numpy as np
 
+import rmgpy.quantity as quantity
 from rmgpy.chemkin import writeKineticsEntry
 from rmgpy.exceptions import InvalidMicrocanonicalRateError, ModifiedStrongCollisionError, PressureDependenceError
 from rmgpy.kinetics import Chebyshev, PDepArrhenius
 from rmgpy.reaction import Reaction
 from rmgpy.kinetics.tunneling import Wigner, Eckart
 from rmgpy.data.kinetics.library import LibraryReaction
```
Comment (amarkpayne) [inline arkane/pdep.py]: Move rmgpy.data in between rmgpy.chemkin and rmgpy.exceptions

File: arkane/statmech.py [hunk 16]
```diff
@@ -39,6 +39,7 @@
 import os.path
 
 import numpy as np
+import matplotlib.pyplot as plt
```
Comment (amarkpayne) [inline arkane/statmech.py]: Swap these

File: arkane/kinetics.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (amarkpayne) [reply arkane/kinetics.py]: Yes, sorry I had a typo in my first comment. Let's leave these in for now and then clean this up in the PEP-8 PR across all files

**Comment by alongd** [comment]
OK, let me know when to squash

File: arkane/common.py [hunk 17]
```diff
@@ -325,23 +324,23 @@ def is_pdep(jobList):
     return False
 
 
-def check_conformer_energy(Vlist, path):
+def check_conformer_energy(v_list, path):
```
Comment (amarkpayne) [inline arkane/common.py]: revert this change for now

File: arkane/explorerTest.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (amarkpayne) [reply arkane/explorerTest.py]: This one is fine for now to leave as is

**Comment by alongd** [comment]
OK, squashed. I'll rebase after `amp_py3_pr` gets merged.
Thanks for the comments!

File: arkane/statmech.py [hunk 18]
```diff
@@ -34,36 +34,35 @@
...
-import math
-import numpy as np
 import logging
+import math
+import os.path
 
-from rdkit.Chem import GetPeriodicTable
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (amarkpayne) [inline arkane/statmech.py]: Delete this duplicated import before you rebase next time

File: arkane/gaussian.py [hunk 19]
```diff
@@ -33,16 +28,23 @@
...
+
 import logging
+import math
 import os.path
 
+import numpy as np
+
 import rmgpy.constants as constants
-from rmgpy.statmech import IdealGasTranslation, NonlinearRotor, LinearRotor, HarmonicOscillator, Conformer
 from rmgpy.exceptions import InputError
```
Comment (amarkpayne) [inline arkane/gaussian.py]: I don't think this import statement is needed anymore, get rid of it during the rebase

File: arkane/thermo.py (hunk 7 above)
Comment (amarkpayne) [inline arkane/thermo.py]: Get rid of this duplicate during the rebase

**Comment by alongd** [comment]
Thanks!

**Comment by mliu49** [comment]
I think we got the PR dependence mixed up. The temp fix which conflicts with #1686 is in #1696.

I think this PR is safe to merge.
